TeamCode/src/main/java/org/firstinspires/ftc/teamcode/auton/ScoringAlgorithm.java

```diff
@@ -53,19 +59,37 @@ private void scoring() {
 
         setJointPos();
 
+
         while (pixelsInClaw) {
             robot.<Servo>get("clawServo").setPosition(robot.<Servo>get("clawServo").getPosition() - 0.125);
-        }
 
+        }
         resetPos();
 
-        telemetry.addData("Scoring Status", "Scored");
+    }
+
+    public void setViperSlidePos() {
+        telemetry.addData("Scoring Status", "Setting VS Pos");
+        // Setting ViperSlidePoses based on Targets.
+        while (Math.abs(robot.<DcMotor>get("leftViperSlideMotor").getCurrentPosition() - viperSlideTargetPos) > viperSlideTolerance) {
+
+                robot.<DcMotor>get("rightViperSlideMotor").setPower(0.4);
+                robot.<DcMotor>get("leftViperSlideMotor").setPower(0.4);
+
+            telemetry.addData("VS right Pos", robot.<DcMotor>get("rightViperSlideMotor").getCurrentPosition());
+            telemetry.addData("VS left Pos", robot.<DcMotor>get("leftViperSlideMotor").getCurrentPosition());
+        }
+
+        robot.<DcMotor>get("rightViperSlideMotor").setPower(0);
+        robot.<DcMotor>get("leftViperSlideMotor").setPower(0);
+
+        telemetry.addData("Scoring Status", "VS Pos Set");
     }
 
     public void setArmPos() {
         telemetry.addData("Scoring Status", "Setting Arm Pos");
         // Setting ArmPos based on Target.
-        while (robot.<DcMotor>get("armMotor").getCurrentPosition() != armTargetPos) {
+        while (Math.abs(robot.<DcMotor>get("armMotor").getCurrentPosition() - armTargetPos) > armTolerance) {
```

Comment:
there's a chance you might want to add a condition to catch if it would ever miss your entire section. you can solve this by either changing the while condition to be `while (robot.<DcMotor>get("armMotor").getCurrentPosition < armTargetPos){}` which would work well to catch it as soon as it passes that threshold

however, if you're more concerned with accuracy and willing to sacrifice some time, you can do something that's basically inside of your while loop that checks `if (robot.<DcMotor>get("armMotor").getCurrentPosition < armTargetPos - armTolerance){}` and add an else if to check `else if (robot.<DcMotor>get("armMotor").getCurrentPosition > armTargetPos + armTolerance){}` which would basically let it self correct

sorry, missed this last time i looked through